### PR TITLE
uorb: introduce SubscriptionMultiArray for working with multi-instances

### PR DESCRIPTION
**Pull request by dagar**
This is a convenient replacement for the growing (ugly) pattern of working with multiple (often all) possible instances of a multi-publication.

### Before
``` C++

	uORB::Subscription distance_sensor_subs[ORB_MULTI_MAX_INSTANCES] {
		{ORB_ID(distance_sensor), 0},
		{ORB_ID(distance_sensor), 1},
		{ORB_ID(distance_sensor), 2},
		{ORB_ID(distance_sensor), 3}
	};

	for (int i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
		if (distance_sensor_subs[i].updated()) {
			// do whatever
		}
	}

```

### After
``` C++
	uORB::SubscriptionMultiArray<distance_sensor_s> distance_sensor_subs{ORB_ID(distance_sensor)};

	for (auto& dist_sens_sub : distance_sensor_subs) {
		if (dist_sens_sub.updated()) {
			// do whatever
		}
	}
```

This also makes it a bit easier to start increasing (or decreasing) `ORB_MULTI_MAX_INSTANCES` as needed.

### CODE REVIEW

File: src/modules/mavlink/mavlink_messages.cpp [hunk 1]
```diff
@@ -2994,10 +2960,10 @@ class MavlinkStreamVibration : public MavlinkStream
 			}
 
 			// accel 0, 1, 2 cumulative clipping
-			for (int i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+			for (int i = 0; i < _vehicle_imu_status_subs.size(); i++) {
```
Comment (dagar) [inline src/modules/mavlink/mavlink_messages.cpp:2963]:
```suggestion
			for (int i = 0; i < 3; i++) {
```

Comment (dagar) [reply src/modules/mavlink/mavlink_messages.cpp:2963]: Fixed.

**Comment by dagar** [comment]
Logger changes for multi-instance. https://github.com/PX4/Firmware/pull/15647

File: src/modules/mavlink/mavlink_messages.cpp [hunk 2]
```diff
@@ -2994,10 +2946,10 @@ class MavlinkStreamVibration : public MavlinkStream
 			}
 
 			// accel 0, 1, 2 cumulative clipping
-			for (int i = 0; i < ORB_MULTI_MAX_INSTANCES; i++) {
+			for (int i = 0; i < 3; i++) {
```
Comment (bkueng) [inline src/modules/mavlink/mavlink_messages.cpp:2949]:
```suggestion
			for (int i = 0; i < math::min(3, _vehicle_imu_status_subs.size()); i++) {
```

**Review by bkueng — APPROVED** [review]
Much nicer

**Pull request merged**